rotkehlchen/tests/unit/test_price_historian.py

```diff
@@ -0,0 +1,233 @@
+        data_directory=MagicMock(spec=Path),
+        cryptocompare=MagicMock(spec=Cryptocompare),
+        coingecko=MagicMock(spec=Coingecko),
+        oracle_order=oracle_order,
+    )
+    return price_historian
+
+
+def test_all_common_methods_implemented():
+    """Test all historical price oracles implement the expected methods.
+    """
+    for oracle in DEFAULT_HISTORICAL_PRICE_ORACLE_ORDER:
+        if oracle == HistoricalPriceOracle.COINGECKO:
+            instance = Coingecko
+        elif oracle == HistoricalPriceOracle.CRYPTOCOMPARE:
+            instance = Cryptocompare
+        else:
+            raise AssertionError(
+                f'Unexpected historical price oracle: {oracle}. Update this test',
+            )
+
+        # Check 'rate_limited_in_last' method exists
+        assert hasattr(instance, 'rate_limited_in_last')
+        assert callable(instance.rate_limited_in_last)
+        # Check 'query_historical_price' method exists
+        assert hasattr(instance, 'query_historical_price')
+        assert callable(instance.query_historical_price)
+
+
+def test_oracle_instances_default_order(price_historian):
+    expected_oracle_instances = [
+        (HistoricalPriceOracle.CRYPTOCOMPARE, price_historian._cryptocompare),
+        (HistoricalPriceOracle.COINGECKO, price_historian._coingecko),
+    ]
+    assert price_historian._oracle_instances == expected_oracle_instances
+
+
+@pytest.mark.parametrize('oracle_order', [
+    [HistoricalPriceOracle.COINGECKO, HistoricalPriceOracle.CRYPTOCOMPARE],
+])
+def test_oracle_instances_custom_order(price_historian):
+    expected_oracle_instances = [
+        (HistoricalPriceOracle.COINGECKO, price_historian._coingecko),
+        (HistoricalPriceOracle.CRYPTOCOMPARE, price_historian._cryptocompare),
+    ]
+    assert price_historian._oracle_instances == expected_oracle_instances
+
+
+def test_fiat_to_fiat(price_historian):
+    """Test the price is returned via exchangerates API when requesting the
+    historical price from fiat to fiat.
+    """
+    expected_price = Price(FVal('1.25'))
+    mock_inquirer = MagicMock()
+    mock_inquirer.query_historical_fiat_exchange_rates.return_value = expected_price
+
+    with patch('rotkehlchen.history.price.Inquirer', return_value=mock_inquirer):
+        price = price_historian.query_historical_price(
+            from_asset=A_USD,
+            to_asset=A_GBP,
+            timestamp=Timestamp(1611595466),
+        )
+
+    assert price == expected_price
+    assert mock_inquirer.query_historical_fiat_exchange_rates.call_count == 1
+
+
+def test_fiat_to_fiat_no_price_found_exception(price_historian):
+    """Test NoPriceForGivenTimestamp is raised when requesting the historical
+    price from fiat to fiat and:
+      - Inquirer returns None.
+      - The other oracles don't support fiat to fiat.
+
+    FakeHistoricalPriceOracle sets 'has_fiat_to_fiat' to False.
+    """
+    class FakeHistoricalPriceOracle(Enum):
```

Comment:
If you find a class within a test ugly, I can try to fake it in another way. Same for the test below.
I wasn't sure how many times I was gonna need and with which properties, so I opted for make it work first.

Reply:
All this to test the thing I told you to at the end remove.

 And yes this test is super ugly and lots of unneeded code. You can just mock `Inquirer().query_historical_fiat_exchange_rates` alone. Just like we do with so many other functions around the tests.

Reply:
I was mocking the `has_fiat_to_fiat` properties not the inquirer function. I've just removed the whole oracle properties abstraction and deleted the tests.